versioning/BRM/dbrm.cpp

```diff
@@ -4453,6 +4454,97 @@ void DBRM::deleteAISequence(uint32_t OID)
   }
 }
 
+void DBRM::addToLBIDList(uint32_t sessionID, vector<LBID_t>& lbidList)
+{
+  boost::shared_ptr<execplan::CalpontSystemCatalog> systemCatalogPtr =
+    execplan::CalpontSystemCatalog::makeCalpontSystemCatalog(sessionID);
+
+  std::unordered_map<execplan::CalpontSystemCatalog::OID,
+    std::unordered_map<execplan::CalpontSystemCatalog::OID,
+      std::vector<struct BRM::EMEntry>>> extentMap;
+
+  int err = 0;
+
+  std::unordered_set<LBID_t> lbidSet;
+  std::unordered_set<execplan::CalpontSystemCatalog::OID> tableOidSet;
+
+  for (const auto i : lbidList)
+  {
+    lbidSet.insert(i);
+    execplan::CalpontSystemCatalog::OID oid;
+    uint16_t dbRoot, segmentNum;
+    uint32_t partitionNum, fbo;
+    err = lookupLocal(i, 0, false, oid, dbRoot, partitionNum, segmentNum, fbo);
+
+    if (err)
+    {
+      ostringstream os;
+      std::string errorMsg;
+      BRM::errString(err, errorMsg);
+      os << "lookupLocal from extent map error encountered while looking up lbid " << i
+         << " and error code is " << err << " with message " << errorMsg;
+      throw runtime_error(os.str());
+    }
+
+    execplan::CalpontSystemCatalog::OID tableOid =
+      systemCatalogPtr->isAUXColumnOID(oid);
+
+    if (tableOid >= 3000)
+    {
+      if (tableOidSet.find(tableOid) == tableOidSet.end())
+      {
+        tableOidSet.insert(tableOid);
+        execplan::CalpontSystemCatalog::TableName tableName =
+          systemCatalogPtr->tableName(tableOid);
+        execplan::CalpontSystemCatalog::RIDList tableColRidList =
+          systemCatalogPtr->columnRIDs(tableName);
+
+        for (unsigned j = 0; j < tableColRidList.size(); j++)
+        {
+          auto objNum = tableColRidList[j].objnum;
+          err = getExtents(objNum, extentMap[tableOid][objNum]);
+
+          if (err)
+          {
+            ostringstream os;
+            os << "BRM lookup error. Could not get extents for OID " << objNum;
+            throw runtime_error(os.str());
+          }
+        }
+      }
+
+      uint32_t extentNumAux = fbo / (execplan::AUX_COL_WIDTH * 1024);
+
+      for (auto iter = extentMap[tableOid].begin(); iter != extentMap[tableOid].end(); iter++)
+      {
+        const std::vector<struct BRM::EMEntry>& extents = iter->second;
+
+        for (const struct BRM::EMEntry& extent : extents)
+        {
+          uint32_t extentNum = extent.blockOffset / (extent.range.size * 1024);
+
+          if (dbRoot == extent.dbRoot && partitionNum == extent.partitionNum &&
+              segmentNum == extent.segmentNum && extentNumAux == extentNum)
+          {
+            lbidSet.insert(extent.range.start);
+            break;
+          }
+        }
+      }
```

Comment:
@denis0x0D This nested for loop above where we add `extent.range.start` to `lbidSet` can potentially already exist in `lbidList`. So if I just create the `lbidSet` only for new lbids and then append those lbids to `lbidList`, there is a possibility of `lbidList`containing duplicate values.

Reply:
ok, thanks got it, clear now.